### ISSUE TITLE
Add `sparse_bidir_logsumexp` that perform sparse LSE reduction independently across rows and columns

### ISSUE THREAD

**Issue by tvercaut** [cai4cai/torchsparsegradutils#88]
## Summary

This is a follow-up and extension of #83 #87 

We should add a new operation called `sparse_bidir_logsumexp` or similar that efficiently computes independent log-sum-exp reductions along both sparse matrix dimensions in a single pass. For clarity, this would create a LSE vector for rows and another one for columns, it is not the same as reducing the entire matrix to a scalar.

The operation is intended for sparse matrices represented in COO/CSR/CSC formats and returns the row-wise and column-wise log-sum-exp reductions simultaneously.

## Motivation

Many algorithms operating on sparse matrices require reductions over both dimensions. This typically requires two separate reduction passes:
```python
row_lse = sparse_logsumexp(A, dim=1)
col_lse = sparse_logsumexp(A, dim=0)
```

This has several drawbacks:
- Traversing the sparse structure twice.
- Duplicating memory traffic.
- Duplicating temporary storage and index processing.
- Potentially constructing two separate autograd graph fragments.
- Missing opportunities for kernel-level optimization.

When both reductions are needed, a dedicated bidirectional primitive could exploit the fact that every nonzero contributes to both outputs.

## Proposed API

```python
sparse_bidir_logsumexp(
    input: Tensor,  # 2-D sparse tensor (COO, CSR, or CSC) - could also be batched
    keepdim: bool = False,
    include_zeros: bool = True,
    output_layout: Literal["tuple", "nested", "padded"] = "tuple",
) -> Tensor | NestedTensor | tuple[Tensor, Tensor]
```

Computes log-sum-exp reductions along both matrix dimensions simultaneously.

The operation is equivalent to:

```python
(
    sparse_logsumexp(input, dim=0, keepdim=keepdim, include_zeros=include_zeros),
    sparse_logsumexp(input, dim=1, keepdim=keepdim, include_zeros=include_zeros),
)
```

but would be implemented more efficiently by traversing the sparse structure only once.

#### `output_layout="tuple"` (default as more pythonic)

```python
col_lse, row_lse = sparse_bidir_logsumexp(
    A,
    output_layout="tuple",
)
```

Returns:

```python
(
    Tensor(shape=(n_cols,)),
    Tensor(shape=(n_rows,)),
)
```

This is the most pythonic and has minimal API complexity.

#### `output_layout="nested"`

```python
out = sparse_bidir_logsumexp(
    A,
    output_layout="nested",
)
```

Returns:

```python
nested_tensor([
    col_lse,
    row_lse,
])
```

This provides a single tensor-like container while preserving variable lengths.

#### `output_layout="padded"`

```python
out = sparse_bidir_logsumexp(
    A,
    output_layout="padded",
)
```

Returns:

```python
Tensor(shape=(2, max(n_rows, n_cols)))
```

This layout would likely be produced directly from the underlying reduction buffers with little or no additional processing, making it a potentially attractive representation for high-performance implementations.

**Comment by tvercaut** [comment]
## Implementation idea
Parallel evaluation via expansion + batched `_scatter_logsumexp`

Both row-wise and column-wise log-sum-exp reductions operate on the same set of nonzero values. Rather than calling `_scatter_logsumexp` twice, we can formulate the problem as a batched reduction and invoke `_scatter_logsumexp` once.

Given column and row reduction indices, construct a batch dimension representing the reduction direction (note that `expand` creates a view rather than allocating twice the memory):

```python
batched_values = values.expand(2, nnz)

batched_scatter_idx = torch.stack([
    col_scatter_idx,  # reduction over rows -> output indexed by columns
    row_scatter_idx,  # reduction over cols -> output indexed by rows
])
```

Conceptually:

```text
batch 0 -> reduce by column index
batch 1 -> reduce by row index
```

and perform a single batched scatter-logsumexp:

```python
out = _scatter_logsumexp(
    batched_values,
    batched_scatter_idx,
    dim=-1,
    dim_size=max(n_rows, n_cols),
)
```

yielding:

```text
out.shape = (2, max(n_rows, n_cols))
```

which corresponds naturally to the proposed `"padded"` output layout.

## Additional note

A generalisation of `_scatter_logsumexp` to handle batched values and indices looks relatively straightforward to implement (moving from `dim=0` to `dim=-1`?) and would also probably help simplifying `_logsumexp_batched` or avoiding its need entirely (it looks like it currently flattens and reshapes).
